linode/ipv6range/framework_resource.go

```diff
@@ -0,0 +1,210 @@
+package ipv6range
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/linode/linodego"
+	"github.com/linode/terraform-provider-linode/linode/helper"
+)
+
+func NewResource() resource.Resource {
+	return &Resource{
+		BaseResource: helper.NewBaseResource(
+			helper.BaseResourceConfig{
+				Name:   "linode_ipv6_range",
+				IDType: types.StringType,
+				Schema: &frameworkResourceSchema,
+			},
+		),
+	}
+}
+
+type Resource struct {
+	helper.BaseResource
+}
+
+func (r *Resource) Create(
+	ctx context.Context,
+	req resource.CreateRequest,
+	resp *resource.CreateResponse,
+) {
+	var data ResourceModel
+	client := r.Meta.Client
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	createOpts := linodego.IPv6RangeCreateOptions{
+		PrefixLength: int(data.PrefixLength.ValueInt64()),
+	}
+
+	if !data.LinodeId.IsNull() && !data.LinodeId.IsUnknown() {
+		createOpts.LinodeID = int(data.LinodeId.ValueInt64())
+	} else if !data.RouteTarget.IsNull() && !data.RouteTarget.IsUnknown() {
+		createOpts.RouteTarget = strings.Split(data.RouteTarget.ValueString(), "/")[0]
+	} else {
+		resp.Diagnostics.AddError(
+			"Failed to create ipv6 range.",
+			"Either linode_id or route_target must be specified.",
+		)
+		return
+	}
+
+	ipv6range, err := client.CreateIPv6Range(ctx, createOpts)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Failed to create ipv6 range.",
+			err.Error(),
+		)
+	}
+
+	data.ID = types.StringValue(strings.TrimSuffix(
+		ipv6range.Range,
+		fmt.Sprintf("/%d", createOpts.PrefixLength)))
+
+	ipv6rangeR, err := client.GetIPv6Range(ctx, data.ID.ValueString())
+	if err != nil {
+		if lerr, ok := err.(*linodego.Error); ok && (lerr.Code == 404 || lerr.Code == 405) {
+			resp.Diagnostics.AddWarning(
+				"IPv6 range does not exist.",
+				fmt.Sprintf("IPv6 range \"%s\" does not exist, removing from state.", data.ID.ValueString()),
+			)
+			resp.State.RemoveResource(ctx)
+			return
+		}
+		resp.Diagnostics.AddError(
+			"Failed to get ipv6 range after creation.",
+			err.Error(),
+		)
```

Comment:
I wonder if we could break this error handling logic out into a helper function since it's also used in the read function.